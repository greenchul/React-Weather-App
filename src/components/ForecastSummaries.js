import React from "react";
import propTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

const ForecastSummaries = ({ forecasts }) => {
  return (
    <div className="forecast-summaries">
      {forecasts.map((singleForecast) => {
        return (
          <ForecastSummary
            key={`${singleForecast.date}`}
            singleForecast={singleForecast}
          />
        );
      })}
    </div>
  );
};

ForecastSummaries.propTypes = {
  forecasts: propTypes.arrayOf(propTypes.shape()).isRequired,
};
// eslint-disable-next-line react/require-default-props

export default ForecastSummaries;
