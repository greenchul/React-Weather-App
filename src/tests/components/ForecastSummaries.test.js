import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description 1",
      icon: "602",
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 2222222,
      description: "Stub description2",
      icon: "800",
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];
  const onForecastSelect = () => {};

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps}
        onForecastSelect={onForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    render(
      <ForecastSummaries
        forecasts={validProps}
        onForecastSelect={onForecastSelect}
      />
    );
    const forecastSummaryInstances = screen.getAllByTestId("forecast-summary");

    expect(forecastSummaryInstances).toHaveLength(2);
  });
});
