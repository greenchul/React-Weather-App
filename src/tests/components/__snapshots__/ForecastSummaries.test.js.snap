// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
         Date: 1111111 
      </div>
      <div
        class="forecast-summary__temperature"
      >
        Max temp: 22
      </div>
      <div
        class="forecast-summary__description"
      >
        Description: Stub description 1
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        Icon: stubIcon1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
         Date: 2222222 
      </div>
      <div
        class="forecast-summary__temperature"
      >
        Max temp: 24
      </div>
      <div
        class="forecast-summary__description"
      >
        Description: Stub description2
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        Icon: stubIcon2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
